lesson07/rest/course-rest-master/cmd/rest-pre/main.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"log"
+	"net/http"
+
+	stub_contacts "github.com/wshaman/contacts-stub"
+	"github.com/wshaman/course-rest/handlers/user"
+
+	"github.com/gorilla/mux"
+)
+
+func main() {
+	router := mux.NewRouter()
+	router.HandleFunc("/api/v0/user", user.List).Methods(http.MethodGet)
+	router.HandleFunc("/api/v0/user", user.Create).Methods(http.MethodPost)
+	router.HandleFunc("/api/v0/user/{id}", user.Delete).Methods(http.MethodDelete)
+	router.HandleFunc("/api/v0/user/{id}", user.View).Methods(http.MethodGet)
+	router.HandleFunc("/api/v0/user/{id}", user.Update).Methods(http.MethodPut)
+	router.HandleFunc("/api/v0/user/", user.Search).Methods(http.MethodPost)
```

Comment:
I would suggest you to change the method and path for this endpoint
First of all this endpoint collides with the user creation one, since technically trailing slash should be dropped. So, a better path would be `/api/v0/search` for example. 
Secondly, the endpoint accepts a query parameter and returns a list of users, which clearly indicates it should be a GET request

Reply:
Hello, path is changed, but how do I change the method? Give me please direction.

Reply:
The method is `http.MethodPost`. You just need to change it to `http.MethodGet`